### PR TITLE
Update Gem file and add test files to our application

### PR DESCRIPTION
**Pull request by francksefu**
[Project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#project-requirements)
Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'posts_counter must be greater than or equal to zero' do
+    subject.posts_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (omar-muhamad) [inline spec/models/user_spec.rb:16]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=equal%20to%20zero.-,Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.,-Need%20a%20big), Please add tests for the `recent_post` methode in the user model.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  it 'comments_counter must be greater than or equal to zero' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments_counter must be greater than or equal to zero' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (omar-muhamad) [inline spec/models/post_spec.rb:26]: - [ ] Same here, Please add tests for `recent_comments` and `update_post_counter` methods in the post model.
- [ ] You should test all models as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=equal%20to%20zero.-,Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.,-Need%20a%20big), Please add tests for `like `and `comment `models too.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @francksefu,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -14,4 +14,21 @@
...
+
+      # Create some posts associated with the user
+      6.times do |i|
+        posts << Post.create(author: user, title: 'love', text: "Post #{i}")
+      end
+
+      recent_post = user1.recent_post
+
+      expect(recent_post.length).to eq(0)
+    end
```
Comment (elisha2kyakpo1) [inline spec/models/user_spec.rb]: - [ ] Kindly test the scenario where the 3 most recent posts are returned, you did well to create 6 posts, but returning 0 will be very difficult for us to know if the method really works.
- [ ] The previous reviewer mentioned that all model methods should be tested, kindly add tests for the likes and comments models and their functions.

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -24,4 +24,40 @@
...
+
+  describe '#recent_comments' do
+    it 'returns up to 5 recent comments' do
+      post = Post.new(author: user, title: 'love', text: 'Hello word')
+      comments = []
+
+      # Create some comments associated with the post
+      10.times do |i|
+        # create(:comment, post: post, text: "Comment #{i}")
```
Comment (elisha2kyakpo1) [inline spec/models/post_spec.rb]: - [ ] Kindly remove this commented code to clean up the codebase.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @francksefu, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

👍 You did well testing the recent comments function in the posts model.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @francksefu 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- Changes requested have been implemented   ✅ 
 🚀  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**